### ISSUE TITLE
using "q" in the debug mode for flutter reports application finished but does not close applicatin

### ISSUE THREAD

**Issue by leemke** [flutter/flutter#80313]
I ran the default application from "flutter create" with "flutter run".
Then during the debut session I keyed the letter "q". It reported 
that the "Application finished" The command line prompt returned.

However, the application window in my environment continued to 
function as if nothing had happened. I was able to close the
application via the "X" system button.

Is this a bug?  Is there a way to detect the end of the application within
the application window, so that the application (i.e. debugger) can tell
the application to shutdown? 

I know if I use the "X" button the debugger will report "lost connection". Suppose the
application is running in background with no visible element on the launch bar? 
Then there would be no way to tell the application to end. Or would it be possible?

![Screenshot 2021-04-12 17 16 16](https://user-images.githubusercontent.com/5303630/114470219-26a99a80-9bb4-11eb-877b-d051833492d0.png)

**Comment by jonahwilliams** [comment]
Can you provide some more information (flutter doctor -v at a minimum) on how you encountered this error and how to reproduce it? Are you talking about a desktop application, an android application, et cetera?

**Comment by leemke** [comment]
here is the information you requested:

~$ flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.0.4, on Linux, locale en_US.UTF-8)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✓] Chrome - develop for the web
[✓] Linux toolchain - develop for Linux desktop
[✓] Android Studio
[✓] Connected device (3 available)

• No issues found!
l:~$ flutter devices
3 connected devices:

Google Pixelbook (mobile) • emulator-5554 • android-x64    • Android 9 (API 28)
Linux (desktop)           • linux         • linux-x64      • Linux
Chrome (web)              • chrome        • web-javascript • Chromium 89.0.4389.114 built on Debian 10.9,
running on Debian 10.9
============================= 

I have installed flutter in the linux/Debian container in Chrome OS.
The installation process was not very easy. It took over a day to get it
correct.  I created a bash script, deleted/reinstalled linux and ran my script.

I installed the latest (?) required support tools.
jdk - 11
flutter - 2.0
android studio 30.0 + plugins
all of the required system packages for desktop support
activated ucb and adb in linux 

I used the command line interface to do a "flutter create"
outside of android studio.
The create instructed me and run "dart migrate --apply-changes"
I changed directory to the lib folder within the project
By default I ran "flutter run". I assume it choose the Mobile device.

I viewed and used the resulting application (in debug mode).
After a few clicks, I returned to the command line window
and pressed the "q" key. The debugger reported that the
application had ended. However, the application window
was still open and active. I could continue to click and 
update the count. I pressed the "X" button and it closed
its application window.

I am new to flutter. I do not know if I installed everything correctly.
It appears to work correctly from what I can tell.

I can rerun the app. I have not changed anything since I created it.
If I can provide any additional information please let me know

**Comment by leemke** [comment]
I see the message about the "old version". I reused the project name from one of the
project I used to build and test the flutter install, before I wiped out linux and rebuilt
everything from scratch.   The "p2" project had an icon in the launcher application list.
I have since, uninstalled that icon.  I do not know how the icon was added to my
application list.  

I have now rerun the test and everything works correctly.  

I assumed that the flutter application only existed in the linux container. By deleting and
reinstalling linux I assumed all previous changes were erased.  Evidently, this is not
the case.  

Thanks for your support. I am still learning about flutter and chrome os.
